src/researchhub_document/hot_score_mixin.py

```diff
@@ -0,0 +1,142 @@
+    # to see a breakdown of how the hot score is calculated for a given document
+    def calculate_hot_score_v2(self, should_save=False):
+        DISCUSSION_VOTE_WEIGHT = 2
+        DOCUMENT_VOTE_WEIGHT = 1
+        hot_score = 0
+        doc = self.get_document()
+
+        # Init debug
+        debug_obj = {
+            'unified_doc_id': self.id,
+            'inner_doc_id': doc.id,
+            'document_type': self.document_type,
+            'DISCUSSION_VOTE_WEIGHT': DISCUSSION_VOTE_WEIGHT,
+            'DOCUMENT_VOTE_WEIGHT': DOCUMENT_VOTE_WEIGHT,
+        }
+
+        # Doc vote score
+        if self.document_type.upper() == 'PAPER':
+            doc_vote_net_score = doc.calculate_score(ignore_twitter_score=True)
+        else:
+            doc_vote_net_score = doc.calculate_score()
+
+        doc_vote_time_score = self._calc_vote_score(doc.votes.all())
+        doc_vote_score = doc_vote_net_score * doc_vote_time_score * DOCUMENT_VOTE_WEIGHT
+        debug_obj['doc_vote_score'] = {'vote_net_score': doc_vote_net_score, 'vote_time_score': doc_vote_time_score, '=doc_vote_score': doc_vote_score}
+
+        # Doc created date score
+        doc_created_score = self._get_date_val(self.created_date)
+        debug_obj['doc_created_score'] = {'created_date': self.created_date, '=doc_created_score': doc_created_score}
+
+        # Doc social media score
+        social_media_score = self._calc_social_media_score()
+        debug_obj['social_media_score'] = {'=social_media_score': social_media_score}
+
+        # Doc discussion vote score
+        discussion_vote_score = 0
+        debug_obj['discussion_vote_score'] = {}
+        for t in doc.threads.filter(is_removed=False):
+            thread_vote_net_score = max(0, t.calculate_score())
+            thread_vote_time_score = self._calc_vote_score(t.votes.all())
+            thread_vote_score = thread_vote_net_score * thread_vote_time_score * DISCUSSION_VOTE_WEIGHT
+            discussion_vote_score += thread_vote_score
+
+            debug_val = {'vote_net_score': thread_vote_net_score, 'vote_time_score': thread_vote_time_score, '=thread_vote_score': thread_vote_score}
+            debug_obj['discussion_vote_score'][f'thread (id:{t.id})'] = debug_val
+            for c in t.comments.filter(is_removed=False):
+                comment_vote_net_score = max(0, c.calculate_score())
+                comment_vote_time_score = self._calc_vote_score(c.votes.all())
+                comment_vote_score = comment_vote_net_score * comment_vote_time_score * DISCUSSION_VOTE_WEIGHT
+                discussion_vote_score += comment_vote_score
+
+                debug_val = {'vote_net_score': comment_vote_net_score, 'vote_time_score': comment_vote_time_score, '=comment_vote_score': comment_vote_score}
+                debug_obj['discussion_vote_score'][f'thread (id:{t.id})'][f'comment (id:{c.id})'] = debug_val
+                for r in c.replies.filter(is_removed=False):
```

Comment:
```suggestion
                for r in c.replies.filter(is_removed=False).iterator():
```

Reply:
what's the point of calling .iterator here? is it to not load everything into memory?